### ISSUE TITLE
3.0 plan

### ISSUE THREAD

**Issue by ChrisDill** [ChrisDill/Raylib-cs#33]
Preparing bindings for the release of raylib 3.0. 

-  **Reduce unsafe with features like Span** I tested it on the Mesh arrays and it makes it easy to use it as a normal array, however I am not sure it is something that should be in the binding code directly.  I am thinking of having a example showing how to use it though.

-  **Improve generator so it can regenerate current bindings.** See #29 for details on this. I have tried different approaches and work on this is ongoing.

The bindings are nearly automated so the hard part is updating all the examples. Help on this is greatly appreciated.

**Comment by ChrisDill** [comment]
Update on progress made.

Issues have been found with 3D here #39 and #35. I am stuck on deciding the best way to handle this issue.

Currently using IntPtr. This allows types like Mesh to Marshall to C# but makes accessing the data more difficult. I have also tried with the newer Span<T> which is nicer although for now I don't think the user should require it.

My main focus at the moment is getting the examples working again and preparing for the next release. I am also learning about continuous integration to see if that can help with this.

**Comment by raysan5** [comment]
@ChrisDill thanks for the update! I plan to release the new version by February. I wanted to finish some things before but I think I'll just release a new version with current state. 😊

**Comment by ChrisDill** [comment]
Another update. More ideas on things etc.

To make things easier, here is a updated task list.

- [ ] Update bindings to raylib 2.6.

- [ ] Update examples to raylib 2.6. See #7.

- [ ]  Include the official release libraries for raylib. Tried with a custom build but trying a different approach this time.

- [ ] Add more details to README on api differences to raylib. Maybe create a seperate wiki with a setup guide.

- [ ] Make extra modules clearer. Looking at moving them to their own class and having the option to load from their own library. Before I tried a custom build that included them with raylib but I ended up deciding against that.

- [ ] Learn more about continuous integration and set it up. Currently testing github actions.

- [ ] Prepare a official release for raylib-cs. A folder with both the binding code and the libraries.

Also note that other tasks like examples using Span and improving the Generator have been put on hold for now.
